### PR TITLE
add job control to oil-native

### PR DESCRIPTION
**Pull request by melvinw**
Just needed a few small fixes to make `builtin_process.py` translatable to get this working

### CODE REVIEW

File: cpp/leaky_stdlib.cc [hunk 1]
```diff
@@ -111,6 +112,10 @@ void execve(Str* argv0, List<Str*>* argv, Dict<Str*, Str*>* environ) {
   InvalidCodePath();
 }
 
+void kill(int pid, int sig) {
+  assert(::kill(pid, sig) == 0);
```
Comment (andychu) [inline cpp/leaky_stdlib.cc:116]: To follow Python, this should check the return code and then

```
 throw Alloc<OSError>(errno);
```

Some of the other functions do this, and we should clean up ones that don't


(Annoying thing: Python 2 sometimes throws IOError and sometimes OSError, and it's not documented.   So I had to look at pyext/posixmodule.c  and read posix_error() to see which one it was! )

**Review by andychu — COMMENTED** [review]
Awesome!   Thanks for doing this

File: cpp/leaky_stdlib.cc (hunk 1 above)
Comment (melvinw) [reply cpp/leaky_stdlib.cc]: Ah nice. This is less aggressive than assert-failing here. Added

**Comment by andychu** [comment]
Thanks!

**Pull request merged**